hal/src/main/native/athena/PDP.cpp

```diff
@@ -62,77 +141,200 @@ HAL_Bool HAL_CheckPDPChannel(int32_t channel) {
   return channel < kNumPDPChannels && channel >= 0;
 }
 
-double HAL_GetPDPTemperature(int32_t module, int32_t* status) {
-  if (!checkPDPInit(module, status)) return 0;
-
-  double temperature;
+double HAL_GetPDPTemperature(HAL_PDPHandle handle, int32_t* status) {
+  PdpStatus3 pdpStatus;
+  int32_t length = 0;
+  uint64_t receivedTimestamp = 0;
 
-  *status = pdp[module]->GetTemperature(temperature);
+  HAL_ReadCANPeriodicPacket(handle, Status3, pdpStatus.data, &length,
+                            &receivedTimestamp, TimeoutMs, StatusPeriodMs,
+                            status);
 
-  return temperature;
+  return pdpStatus.bits.temp * 1.03250836957542 - 67.8564500484966;
```

Comment:
Umm CTRE does not have a comment on this one.  I am going to assume this is the Fx to convert from ADC value to temp that was experimentally found?  Maybe we should ask them.

Reply:
For now I just want it a straight port. We can add docs later if we get them from CTRE.